src/custom/state/swap/extension.ts

```diff
@@ -10,22 +10,26 @@ interface ExtendedTradeParams {
   feeAsCurrency?: CurrencyAmount
 }
 
+export type TradeWithFee = Trade & { inputAmountWithFee: CurrencyAmount }
+
 /**
  * extendExactInTrade
  * @description takes a Uni ExactIn Trade object and returns a custom one with fee adjusted inputAmount
  */
-export function extendExactInTrade(params: Pick<ExtendedTradeParams, 'exactInTrade' | 'typedAmountAsCurrency'>) {
+export function extendExactInTrade(
+  params: Pick<ExtendedTradeParams, 'exactInTrade' | 'typedAmountAsCurrency'>
+): TradeWithFee | null {
   const { exactInTrade, typedAmountAsCurrency } = params
 
   if (!exactInTrade || !typedAmountAsCurrency) return null
 
-  // We need to iverride the Trade object to use different values as we are intercepting initial inputs
+  // We need to override the Trade object to use different values as we are intercepting initial inputs
   // and applying fee. For ExactIn orders, we leave outputAmount as is
   // and only change inputAmount to show the original entry before fee calculation
   return {
     ...exactInTrade,
     inputAmount: typedAmountAsCurrency,
-    inputAmountWithFees: exactInTrade.inputAmount,
+    inputAmountWithFee: exactInTrade.inputAmount,
```

Comment:
I might be wrong,  but I believe `inputAmount` comes from a step before where you deducted the fees. So is it really `inputAmountWithFee` or is it `inputAmountWithoutFee`. I'm confused 🤯

Reply:
`inputAmount` here === user's typed in amount, we can use `inputAmountWithFee` as the amount we pass to the order posting function

the `typedAmountAsCurrency` is simply the user's typed in amount, no maths done onto the amount, just formatting